core/authority-discovery/src/lib.rs

```diff
@@ -296,6 +304,44 @@ where
 		self.address_cache
 			.retain(|peer_id, _addresses| current_authorities.contains(peer_id))
 	}
+
+	fn get_priv_keys_within_authority_set(&mut self) -> Result<Vec<AuthorityPair>> {
```

Comment:
this is a non-trivial function, so it needs some docs.

Reply:
Docs added in https://github.com/paritytech/substrate/pull/3955/commits/89f501888c87e5fabeead46ed6e3bdf09d3410c2.